pkg/agent/agent.go

```diff
@@ -335,7 +335,7 @@ func (a *Agent) setAPIKeyConfigFrom(ctx context.Context, secrets []*kates.Secret
 	// there _should_ only be one secret here, but we're going to loop and check that the object
 	// meta matches what we expect
 	for _, secret := range secrets {
-		if (secret.GetName() == a.agentCloudResourceConfigName || secret.GetName() == cloudConnectTokenDefaultResource) && secret.GetNamespace() == a.agentNamespace {
+		if (secret.GetName() == a.agentCloudResourceConfigName || strings.HasSuffix(secret.GetName(), cloudConnectTokenDefaultSuffix)) && secret.GetNamespace() == a.agentNamespace {
```

Comment:
Is the watcher setup to watch all secrets in the namespace? I was under the impression it selected which secrets to watch for, depending on the `cloudConnectTokenDefaultResource` name.

I'm also thinking there might be multiple secrets and configmap that matches the pattern. We should probably log something informative to the user who might be debugging this like
```
Looking up for cloud connect tokens in 2 Secrets and 1 ConfigMaps
```

Reply:
`fieldSelector` does not support regex, so it would be difficult to get all secrets with a certain suffix

Reply:
I noticed that back when the agent was a part of aes, the configmap with the token had the label `product: aes`; you could match against that and we can have a `product:ambassador-agent` label going forward.

Reply:
I'd be careful tho, as depending on the Secret/ConfigMap creation flow, that label might not be present. It is the case if you are using kubectl in-line cm creation.

Reply:
yeah, I think it might be one of those situations where if a user created the cm that way then this wouldn't really be an issue if they can pass in the resource by name.
It'd be pretty clean if we didn't have to look into every secret and cm in the agent's ns, especially for users who might want to install the agent into `kube-system` or some other critical shared ns.